crates/iota-framework/packages/iota-system/sources/validator_set.move

```diff
@@ -479,7 +640,44 @@ module iota_system::validator_set {
         result
     }
 
-    // ==== getter functions ====
+    /// Called by `iota_system` to derive reference gas price for the new epoch for ValidatorSetV2.
+    /// Derive the reference gas price based on the gas price quote submitted by each committee validator.
+    /// The returned gas price should be greater than or equal to 2/3 of the validators submitted
+    /// gas price, weighted by voting power in the current epoch.
+    public(package) fun derive_reference_gas_price_inner(self: &ValidatorSetV2): u64 {
+        let vs = &self.active_validators;
+        let committee_members = &self.committee_members;
+        let num_validators = vs.length();
+        let num_committee_members = self.committee_members.length();
+        let mut entries = vector[];
+        let mut i = 0;
+        while (i < num_committee_members) {
+            let validator_index = committee_members[i];
+            assert!(
+                validator_index < num_validators,
+                ECommitteeMembersSetCorrupt,
+            );
+
+            let v = &vs[validator_index];
```

Comment:
Is it possible to hide this pattern (getting validator by index including the assert above) somehow, eg. in a macro fun? It is repeated a lot

Reply:
Good point! Replaced it with a function.